### ISSUE TITLE
thread panicked while panicking, while reading py2 syntax

### ISSUE THREAD

**Issue by jayvdb** [indygreg/PyOxidizer#147]
I tried including CPython tests with `exclude_test_modules = false`, and BOOM indeed. Using branch `main`.

```py
Traceback (most recent call last):
  File "/tmp/bytecode-compiler.2TplNMU2Kyfz/bytecodecompiler.py", line 65, in <module>
    code = compile(source, name, 'exec', optimize=optimize_level)
  File "lib2to3.tests.data.bom", line 2
    print "BOM BOOM!"
                    ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("BOM BOOM!")?
```
```rust
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: Empty }', src/libcore/result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
thread 'main' panicked at 'write failed: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }', src/libcore/result.rs:997:5
stack backtrace:
   0:     0x55c68c04369f - std::sys::unix::backtrace::tracing::imp::unwind_backtrace::h2387d995b9073026
                               at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1:     0x55c68c038a97 - std::sys_common::backtrace::print::h73caeaceb668a47d
                               at src/libstd/sys_common/backtrace.rs:71
                               at src/libstd/sys_common/backtrace.rs:59
   2:     0x55c68c0264c3 - std::panicking::default_hook::{{closure}}::h343883257d6a605e
                               at src/libstd/panicking.rs:197
   3:     0x55c68c026250 - std::panicking::default_hook::h896c5bc427faced8
                               at src/libstd/panicking.rs:211
   4:     0x55c68c026bbf - std::panicking::rust_panic_with_hook::h64d2566eb4f32ecc
                               at src/libstd/panicking.rs:474
   5:     0x55c68c026751 - std::panicking::continue_panic_fmt::h0df69e4bd5d06d5b
                               at src/libstd/panicking.rs:381
   6:     0x55c68c026635 - rust_begin_unwind
                               at src/libstd/panicking.rs:308
   7:     0x55c68c058b6c - core::panicking::panic_fmt::h02eac3544c403eac
                               at src/libcore/panicking.rs:85
   8:     0x55c68b9e289d - core::result::unwrap_failed::ha3cefcd67f7f4376
   9:     0x55c68ba456a6 - <pyoxidizer::pyrepackager::bytecode::BytecodeCompiler as core::ops::drop::Drop>::drop::h628fb00f6607e229
  10:     0x55c68ba7bf0b - core::ptr::real_drop_in_place::h110dca9e9d8b5626
  11:     0x55c68ba87a64 - pyoxidizer::pyrepackager::repackage::resolve_python_resources::hee3eec2ed251c63b
  12:     0x55c68ba90726 - pyoxidizer::pyrepackager::repackage::process_config::h070fe1f20940daae
  13:     0x55c68b9b4c1c - pyoxidizer::projectmgmt::build_pyoxidizer_artifacts::h6a72250ab91d4983
  14:     0x55c68b9b5838 - pyoxidizer::projectmgmt::build_project::h1a019c9fc2cdf901
  15:     0x55c68b9b79ce - pyoxidizer::projectmgmt::build::h417b1e511ebfb7d9
  16:     0x55c68b9fb0aa - pyoxidizer::cli::run_cli::he8adb28492905319
  17:     0x55c68b97d921 - pyoxidizer::main::h1df575c8b0c2e787
  18:     0x55c68ba27b52 - std::rt::lang_start::{{closure}}::h991dadd0c6328ce8
  19:     0x55c68c0265d2 - std::panicking::try::do_call::h234aa24f95aba63f
                               at src/libstd/rt.rs:49
                               at src/libstd/panicking.rs:293
  20:     0x55c68c045e69 - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:87
  21:     0x55c68c03e12f - std::rt::lang_start_internal::ha51c422a47a0dc3f
                               at src/libstd/panicking.rs:272
                               at src/libstd/panic.rs:388
                               at src/libstd/rt.rs:48
  22:     0x55c68b97de44 - main
  23:     0x7fb053bdee0a - __libc_start_main
  24:     0x55c68b93c6f9 - _start
                               at ../sysdeps/x86_64/start.S:120
  25:                0x0 - <unknown>
thread panicked while panicking. aborting.
Illegal instruction (core dumped)
```
If the file cant be byte compiled, it should be discarded, or added as a resource instead of as a module.

Trying to load all of the files in the test module is a great sanity check; I used it to find lots of bugs in `pyflakes`.

The final result wont be immediately very useful, as there are lots of byte-compilable .py files in there which need to be accessed as raw files, but I guess we could tell pyoxidizer which files are modules and which are resources.

**Comment by indygreg** [comment]
Yeah, we should handle parse/compilation failures more gracefully. We probably want the behavior to be a config option, as I could see people wanting to fail fast if a `.py` does not compile.

**Comment by jayvdb** [comment]
Lots of 'core' packages contain a py2 compat module which does not compile on py3, and vice-versa, so I would expect it to be ignored by default.  setuptools normally does byte-compiling and it emits the error but otherwise keeps going.
IMO fast-fail could be useful, but it is non-standard, and the extra work that goes with it, is feature functionality and very low priority.

**Comment by jayvdb** [comment]
Perhaps a quick fix is to add `excludes` to packaging rule `stdlib` and maybe default it to the following, which avoids the problems with py37 at least, and can be extended to cover problem files in other CPython tests.
```toml
excludes = [
  "lib2to3.tests.data",
  "test.bad_coding",
  "test.badsyntax_3131",
  "test.badsyntax_future3",
  "test.badsyntax_future4",
  "test.badsyntax_future5",
  "test.badsyntax_future6",
  "test.badsyntax_future7",
  "test.badsyntax_future8",
  "test.badsyntax_future9",
  "test.badsyntax_future10",
  "test.badsyntax_pep3120",
]
```

**Comment by indygreg** [comment]
I believe the "double panic" part of this issue was resolved by the fix mentioned in #334. That leaves the failure to compile bytecode for standard library test modules, which I'll push a fix for shortly.

I'm going to stop short of implementing a feature to make bytecode compilation failures non-fatal. If this is still wanted, please file a new issue, referencing this one.
